main.py

```diff
@@ -0,0 +1,188 @@
+import SoapySDR
+from SoapySDR import *  # SOAPY_SDR_ constants
+import numpy as np  # use numpy for buffers
+import pyfftw
+from matplotlib import pyplot as plt
+import keyboard
+from PyQt5 import QtWidgets
+from PyQt5.QtWidgets import QApplication,QMainWindow
+import sys
+
+def window():
+    app = QApplication(sys.argv)
+    win = QMainWindow()
+    win.setGeometry(200, 200, 300, 300)
+    win.setWindowTitle("soapySpectrum Menu")
+
+    label = QtWidgets.QLabel(win)
+    label.setText("myLabel")
+    label.move(50, 50)
+
+    win.show()
+
+# apply initial settings to HackRF device
+def initializeHackRF(fs, f_rx, bw, gain):
+    sdr.setSampleRate(SOAPY_SDR_RX, 0, fs)
+    sdr.setBandwidth(SOAPY_SDR_RX, 0, bw)
+    sdr.setFrequency(SOAPY_SDR_RX, 0, f_rx)
+    sdr.setGain(SOAPY_SDR_RX, 0, gain)
+
+
+def plotUpdate(ln, sig, frequencies, rxfreq):
+    ln.set_ydata(np.abs(sig))
+    # ln.set_ydata(np.log10(np.abs(sig)))
+    ln.set_xdata((frequencies + rxfreq)/1e6)
+    # plt.ylim(0, 1200)
+    plt.gca().relim()
+    plt.gca().autoscale_view()
+    plt.pause(0.01)
+
+
+def printMenu():
+    print("Choose one from the options:")
+    print("1 - Change RX frequency")
+    print("2 - Enable max hold")
+    print("3 - Disable max hold")
+    print("4 - Enable moving average")
+    print("5 - Disable moving average")
+    print("6 - Change moving average ratio")
+    print("7 - Clear plot")
+    print("8 - Print menu again")
+    print("9 - Quit")
+
+
+# window()
+
+# enumerate devices
+results = SoapySDR.Device.enumerate()
+for result in results: print(result)
+
+# create device instance
+# args can be user defined or from the enumeration result
+args = dict(driver="hackrf")
+sdr = SoapySDR.Device(args)
+
+# query device info
+print(sdr.listAntennas(SOAPY_SDR_RX, 0))
+print(sdr.listGains(SOAPY_SDR_RX, 0))
+# freqs = sdr.getFrequencyRange(SOAPY_SDR_RX, 0)
+# for freqRange in freqs: print(freqRange)
+
+bandwidth = 1.75e6
+samp_rate = 4e6
+rx_freq = 315e6
+buff_len = 512
+RX_gain = 30
+movingAverageRatio = 0.125
+
+initializeHackRF(samp_rate, rx_freq, bandwidth,RX_gain)
+
+# setup a stream (complex floats)
+rxStream = sdr.setupStream(SOAPY_SDR_RX, SOAPY_SDR_CF32)
+print(sdr.getStreamMTU(rxStream))
+sdr.activateStream(rxStream)  # start streaming
+
+# create a re-usable buffer for rx samples
+buff = np.array([0] * buff_len, np.complex64)
+
+# create the plot and the frequency vector
+# freqs = np.fft.fftshift(np.fft.fftfreq(buff_len, d=1/samp_rate))
+freqs = pyfftw.interfaces.numpy_fft.fftshift(pyfftw.interfaces.numpy_fft.fftfreq(buff_len, d=1/samp_rate))
+(line, ) = plt.plot((freqs + rx_freq)/1e6, np.zeros(np.size(freqs)))
+plt.xlabel("Frequency (MHz)")
+plt.ylabel("RSSI")
```

Comment:
this isn't trivial to understand, it seems like you are converting your data to frequencies and than plotting it.
split this section to a technical part and the plot part.
2 functions perhaps. dividing responsibilities makes your codes easier to read, understand and update later.